addons/binding/org.openhab.binding.smhi/src/main/java/org/openhab/binding/smhi/internal/SmhiHandler.java

```diff
@@ -0,0 +1,371 @@
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.eclipse.smarthome.io.net.http.HttpUtil;
+import org.openhab.binding.smhi.internal.model.SmhiData;
+import org.openhab.binding.smhi.internal.model.SmhiParameterTables;
+import org.openhab.binding.smhi.internal.model.SmhiParameters;
+import org.openhab.binding.smhi.internal.model.SmhiTimeSeries;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+/**
+ * The {@link smhiHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Michael Parment - Initial contribution
+ */
+@NonNullByDefault
+public class SmhiHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(SmhiHandler.class);
+    private @Nullable ScheduledFuture<?> refreshJob;
+
+    protected static final String smhiURL = "http://opendata-download-metfcst.smhi.se/api/category/pmp3g/version/2/geotype/point/lon/%s/lat/%s/data.json";
+
+    protected @Nullable PointType location;
+
+    // Timeout for weather data requests.
+    private static final int SMHI_TIMEOUT = 5000;
+
+    private Gson gson = new GsonBuilder().create();
+
+    private String apiRequest = "";
+    private String groupId = "";
+
+    @Nullable
+    private SmhiConfiguration config;
+
+    public SmhiHandler(Thing thing) {
+        super(thing);
+    }
+
+    private SmhiParameterTables parameterTable = new SmhiParameterTables();
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            queryApiAndUpdateChannels();
+        }
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing SMHI weather handler.");
+
+        config = getConfigAs(SmhiConfiguration.class);
+
+        if (StringUtils.trimToNull(config.location) == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "The 'location' parameter must be configured.");
+            return;
+        }
+
+        try {
+            location = new PointType(config.location);
+        } catch (IllegalArgumentException e) {
+            location = null;
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "The 'location' parameter could not be split into latitude and longitude.");
+            return;
+        }
+
+        DecimalFormat df = new DecimalFormat("##.######");
+        DecimalFormatSymbols custom = new DecimalFormatSymbols();
+        custom.setDecimalSeparator('.');
+        df.setDecimalFormatSymbols(custom);
+
+        String[] parts = config.location.split(",");

```

Comment:
You should either check with a regex if the format is ok or check if the split is successfull and both parts are existing and try-catch the double conversion. I would assume that the regex is much easier, probably something like `\d{2}\.\d+,\d{1,2}\.\d+` would be enough.